# Drug Data

**By:** Carol Cheng

**Link to Dataset:** https://corgis-edu.github.io/corgis/python/drugs/

**Description of Data:** This dataset is about substance abuse among different age groups and states and includes data on drugs such as cigarettes, marijuana, cocaine, and alcohol. The data ranges from 2002 to 2018. Both totals (in thousands of people) and rates (as a percentage of the population) are given.

In [24]:
import drugs
report = drugs.get_report()

In [25]:
help(drugs)

Help on module drugs:

NAME
    drugs

DESCRIPTION
    Hello! Thank you for downloading a CORGIS library. However, you do not
    need to open this file. Instead you should make your own Python file and
    add the following line:
    
    import drugs
    
    Then just place the files you downloaded alongside it.

FUNCTIONS
    get_report()
        Retrieves all of the report.

DATA
    __all__ = ['get_report']

FILE
    /home/jovyan/dh140/drugs.py




In [36]:
dir(drugs)

['DatasetException',
 '_Constants',
 '__all__',
 '__builtins__',
 '__cached__',
 '__doc__',
 '__file__',
 '__loader__',
 '__name__',
 '__package__',
 '__spec__',
 '_os',
 '_pickle',
 '_tifa_definitions',
 'get_report']

We dont know what 'report' is yet, so we use this command to find out

In [9]:
type(report)

list

To see the length of variable, we use this command

In [4]:
len(report)

867

In [26]:
report

[{'State': 'Alabama',
  'Year': 2002,
  'Population': {'12-17': 380805, '18-25': 499453, '26+': 2812905},
  'Totals': {'Alcohol': {'Use Disorder Past Year': {'12-17': 18,
     '18-25': 68,
     '26+': 138},
    'Use Past Month': {'12-17': 57, '18-25': 254, '26+': 1048}},
   'Tobacco': {'Cigarette Past Month': {'12-17': 52, '18-25': 196, '26+': 728},
    'Use Past Month': {'12-17': 63, '18-25': 226, '26+': 930}},
   'Illicit Drugs': {'Cocaine Used Past Year': {'12-17': 6,
     '18-25': 27,
     '26+': 49}},
   'Marijuana': {'New Users': {'12-17': 20, '18-25': 18, '26+': 2},
    'Used Past Month': {'12-17': 24, '18-25': 62, '26+': 73},
    'Used Past Year': {'12-17': 49, '18-25': 119, '26+': 141}}},
  'Rates': {'Alcohol': {'Use Disorder Past Year': {'12-17': 0.048336,
     '18-25': 0.13649,
     '26+': 0.049068},
    'Use Past Month': {'12-17': 0.150033, '18-25': 0.509551, '26+': 0.372703}},
   'Tobacco': {'Cigarette Past Month': {'12-17': 0.136906,
     '18-25': 0.392404,
     '26+': 0.

To see the first element of the variable, we use this command

In [23]:
report[0]

{'State': 'Alabama',
 'Year': 2002,
 'Population': {'12-17': 380805, '18-25': 499453, '26+': 2812905},
 'Totals': {'Alcohol': {'Use Disorder Past Year': {'12-17': 18,
    '18-25': 68,
    '26+': 138},
   'Use Past Month': {'12-17': 57, '18-25': 254, '26+': 1048}},
  'Tobacco': {'Cigarette Past Month': {'12-17': 52, '18-25': 196, '26+': 728},
   'Use Past Month': {'12-17': 63, '18-25': 226, '26+': 930}},
  'Illicit Drugs': {'Cocaine Used Past Year': {'12-17': 6,
    '18-25': 27,
    '26+': 49}},
  'Marijuana': {'New Users': {'12-17': 20, '18-25': 18, '26+': 2},
   'Used Past Month': {'12-17': 24, '18-25': 62, '26+': 73},
   'Used Past Year': {'12-17': 49, '18-25': 119, '26+': 141}}},
 'Rates': {'Alcohol': {'Use Disorder Past Year': {'12-17': 0.048336,
    '18-25': 0.13649,
    '26+': 0.049068},
   'Use Past Month': {'12-17': 0.150033, '18-25': 0.509551, '26+': 0.372703}},
  'Tobacco': {'Cigarette Past Month': {'12-17': 0.136906,
    '18-25': 0.392404,
    '26+': 0.258844},
   'Use Past 

In [24]:
report[0]['State']

'Alabama'

To see the keys within 'Rates' we use this command

In [11]:
report[0]['Rates'].keys()

dict_keys(['Alcohol', 'Tobacco', 'Illicit Drugs', 'Marijuana'])

After seeing the keys within 'Rates' we can look at the specific rates of the different keys of a specific state and year

In [43]:
report[0]['Rates']

{'Alcohol': {'Use Disorder Past Year': {'12-17': 0.048336,
   '18-25': 0.13649,
   '26+': 0.049068},
  'Use Past Month': {'12-17': 0.150033, '18-25': 0.509551, '26+': 0.372703}},
 'Tobacco': {'Cigarette Past Month': {'12-17': 0.136906,
   '18-25': 0.392404,
   '26+': 0.258844},
  'Use Past Month': {'12-17': 0.166578, '18-25': 0.451976, '26+': 0.330659}},
 'Illicit Drugs': {'Cocaine Used Past Year': {'12-17': 0.016556,
   '18-25': 0.054892,
   '26+': 0.017513}},
 'Marijuana': {'New Users': {'12-17': 0.059732,
   '18-25': 0.062325,
   '26+': 0.000914},
  'Used Past Month': {'12-17': 0.063662, '18-25': 0.124672, '26+': 0.025967},
  'Used Past Year': {'12-17': 0.127535, '18-25': 0.23788, '26+': 0.050275}}}

Is cocaine use in the past year higher in ages 26+ in Alabama in 2002? (conditional expression)

In [40]:
report[0]['Rates']['Illicit Drugs']['Cocaine Used Past Year']['12-17']  < report[0]['Rates']['Illicit Drugs']['Cocaine Used Past Year']['26+']

True

In [41]:
report[0]['Rates']['Illicit Drugs']['Cocaine Used Past Year']['18-25']  < report[0]['Rates']['Illicit Drugs']['Cocaine Used Past Year']['26+']

False

Total cocaine use in the past year in Alabama from all ages in 2002 (mathematical operation)

In [42]:
report[0]['Rates']['Illicit Drugs']['Cocaine Used Past Year']['12-17']  + report[0]['Rates']['Illicit Drugs']['Cocaine Used Past Year']['18-25'] + report[0]['Rates']['Illicit Drugs']['Cocaine Used Past Year']['26+']

0.08896100000000001

To see the total alcohol use disorder in the past year within the 12-17 age ranges in 2002, we can use a for loop to find every single time 2002 is mentioned and find the correlating state and alcohol use rates.

In [29]:
for i in report:

    if i['Year'] == 2002:
        print(i['Year'],i['State'],i ['Totals']['Alcohol']['Use Disorder Past Year']['12-17'])

2002 Alabama 18
2002 Alaska 4
2002 Arizona 36
2002 Arkansas 14
2002 California 173
2002 Colorado 26
2002 Connecticut 16
2002 Delaware 4
2002 District of Columbia 1
2002 Florida 73
2002 Georgia 36
2002 Hawaii 7
2002 Idaho 10
2002 Illinois 66
2002 Indiana 34
2002 Iowa 19
2002 Kansas 12
2002 Kentucky 17
2002 Louisiana 24
2002 Maine 6
2002 Maryland 26
2002 Massachusetts 37
2002 Michigan 57
2002 Minnesota 32
2002 Mississippi 10
2002 Missouri 31
2002 Montana 9
2002 Nebraska 14
2002 Nevada 13
2002 New Hampshire 9
2002 New Jersey 37
2002 New Mexico 14
2002 New York 87
2002 North Carolina 39
2002 North Dakota 6
2002 Ohio 57
2002 Oklahoma 17
2002 Oregon 17
2002 Pennsylvania 55
2002 Rhode Island 6
2002 South Carolina 17
2002 South Dakota 7
2002 Tennessee 23
2002 Texas 118
2002 Utah 11
2002 Vermont 3
2002 Virginia 38
2002 Washington 31
2002 West Virginia 9
2002 Wisconsin 35
2002 Wyoming 3


We can also use the for loop command to create a function in which we are able to compare the use of cigarettes in the past month in Alabama within the 18-25 age group for any years

In [36]:
def statediff(state1, state2):
    for i in report:
        if i['Year'] == 2010:
            if i['State'] == state1:
                numstate1 = i['Totals']['Marijuana']['New Users']['12-17']
            elif i['State'] == state2:
                numstate2 = i['Totals']['Marijuana']['New Users']['12-17']
    if numstate2 > numstate1:
        print('New Users for Marijuana in 2010 for', state2, 'was greater than for', state1)
    elif numstate1 > numstate2 : 
        print('New Users for Marijuana in 2010 for', state2, 'was less than for', state1)
    else: 
        print('New Users for Marijuana in 2010 for', state2, 'was  equal to that of', state1)

In [37]:
statediff('California','Wyoming')

New Users for Marijuana in 2010 for Wyoming was less than for California


In [38]:
statediff('Nevada','Oregon')

New Users for Marijuana in 2010 for Oregon was greater than for Nevada


Since the variable 'drugs' has 867 elements, we can use the pandas library to clean it just for fun

In [23]:
import pandas as pd

In [30]:
df = pd.read_csv('drugs.csv')

In [31]:
df

State  Year  Population.12-17  Population.18-25  Population.26+  \
0          Alabama  2002            380805            499453         2812905   
1           Alaska  2002             69400             62791          368460   
2          Arizona  2002            485521            602265         3329482   
3         Arkansas  2002            232986            302029         1687337   
4       California  2002           3140739           3919577        21392421   
..             ...   ...               ...               ...             ...   
862       Virginia  2018            629725            869285         5581639   
863     Washington  2018            545968            738052         5065742   
864  West Virginia  2018            124659            174236         1235448   
865      Wisconsin  2018            442510            615930         3861670   
866        Wyoming  2018             44908             57395          377084   

     Totals.Alcohol.Use Disorder Past Year.12-17  \
0                                             18   
1                                              4   
2                                             36   
3                                             14   
4                                            173   
..                                           ...   
862                                           10   
863                                           11   
864                                            2   
865                                            8   
866                                            1   

     Totals.Alcohol.Use Disorder Past Year.18-25  \
0                                             68   
1                                             12   
2                                            117   
3                                             53   
4                                            581   
..                                           ...   
862                                           82   
863                                           75   
864                                           15   
865                                           74   
866                                            7   

     Totals.Alcohol.Use Disorder Past Year.26+  \
0                                          138   
1                                           27   
2                                          258   
3                                          101   
4                                         1298   
..                                         ...   
862                                        267   
863                                        282   
864                                         46   
865                                        217   
866                                         22   

     Rates.Alcohol.Use Disorder Past Year.12-17  \
0                                      0.048336   
1                                      0.061479   
2                                      0.073819   
3                                      0.061457   
4                                      0.055109   
..                                          ...   
862                                    0.015199   
863                                    0.020069   
864                                    0.017428   
865                                    0.018720   
866                                    0.023695   

     Rates.Alcohol.Use Disorder Past Year.18-25  ...  \
0                                      0.136490  ...   
1                                      0.187891  ...   
2                                      0.193626  ...   
3                                      0.175913  ...   
4                                      0.148312  ...   
..                                          ...  ...   
862                                    0.094035  ...   
863                                    0.101770  ...   
864                                    0.087561  ...   
865                         

There are a bunch of different functions we can test out within pandas 

In [32]:
df.min()

State                                                 Alabama
Year                                                     2002
Population.12-17                                        30551
Population.18-25                                        57395
Population.26+                                         310110
Totals.Alcohol.Use Disorder Past Year.12-17                 0
Totals.Alcohol.Use Disorder Past Year.18-25                 6
Totals.Alcohol.Use Disorder Past Year.26+                  19
Rates.Alcohol.Use Disorder Past Year.12-17           0.012143
Rates.Alcohol.Use Disorder Past Year.18-25           0.071218
Rates.Alcohol.Use Disorder Past Year.26+             0.033838
Totals.Alcohol.Use Past Month.12-17                         3
Totals.Alcohol.Use Past Month.18-25                        32
Totals.Alcohol.Use Past Month.26+                         167
Rates.Alcohol.Use Past Month.12-17                   0.050921
Rates.Alcohol.Use Past Month.18-25                   0.303587
Rates.Al

In [33]:
df.max()

State                                                 Wyoming
Year                                                     2018
Population.12-17                                      3293484
Population.18-25                                      4469106
Population.26+                                       25917724
Totals.Alcohol.Use Disorder Past Year.12-17               204
Totals.Alcohol.Use Disorder Past Year.18-25               717
Totals.Alcohol.Use Disorder Past Year.26+                1586
Rates.Alcohol.Use Disorder Past Year.12-17           0.112131
Rates.Alcohol.Use Disorder Past Year.18-25           0.272941
Rates.Alcohol.Use Disorder Past Year.26+              0.11272
Totals.Alcohol.Use Past Month.12-17                       540
Totals.Alcohol.Use Past Month.18-25                      2639
Totals.Alcohol.Use Past Month.26+                       14513
Rates.Alcohol.Use Past Month.12-17                   0.252281
Rates.Alcohol.Use Past Month.18-25                   0.760744
Rates.Al

In [34]:
df.mean()

/tmp/ipykernel_88/3698961737.py:1: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.
  df.mean()


Year                                                 2.010000e+03
Population.12-17                                     4.897141e+05
Population.18-25                                     6.588800e+05
Population.26+                                       3.874155e+06
Totals.Alcohol.Use Disorder Past Year.12-17          1.922491e+01
Totals.Alcohol.Use Disorder Past Year.18-25          9.448212e+01
Totals.Alcohol.Use Disorder Past Year.26+            2.241476e+02
Rates.Alcohol.Use Disorder Past Year.12-17           4.094951e-02
Rates.Alcohol.Use Disorder Past Year.18-25           1.511800e-01
Rates.Alcohol.Use Disorder Past Year.26+             5.976760e-02
Totals.Alcohol.Use Past Month.12-17                  6.580161e+01
Totals.Alcohol.Use Past Month.18-25                  3.930185e+02
Totals.Alcohol.Use Past Month.26+                    2.124662e+03
Rates.Alcohol.Use Past Month.12-17                   1.368088e-01
Rates.Alcohol.Use Past Month.18-25                   6.082838e-01
Rates.Alco

In [35]:
df.describe()

Year  Population.12-17  Population.18-25  Population.26+  \
count   867.000000      8.670000e+02      8.670000e+02    8.670000e+02   
mean   2010.000000      4.897141e+05      6.588800e+05    3.874155e+06   
std       4.901807      5.637959e+05      7.559898e+05    4.320776e+06   
min    2002.000000      3.055100e+04      5.739500e+04    3.101100e+05   
25%    2006.000000      1.315405e+05      1.742935e+05    1.027871e+06   
50%    2010.000000      3.396850e+05      4.562400e+05    2.698757e+06   
75%    2014.000000      5.410950e+05      7.468080e+05    4.509094e+06   
max    2018.000000      3.293484e+06      4.469106e+06    2.591772e+07   

       Totals.Alcohol.Use Disorder Past Year.12-17  \
count                                   867.000000   
mean                                     19.224913   
std                                      25.290532   
min                                       0.000000   
25%                                       5.000000   
50%                                      11.000000   
75%                                      24.000000   
max                                     204.000000   

       Totals.Alcohol.Use Disorder Past Year.18-25  \
count                                   867.000000   
mean                                     94.482122   
std                                     108.272774   
min                                       6.000000   
25%                                      26.000000   
50%                                      64.000000   
75%                                     119.500000   
max                                     717.000000   

       Totals.Alcohol.Use Disorder Past Year.26+  \
count                                 867.000000   
mean                                  224.147636   
std                                   254.022992   
min                                    19.000000   
25%                                    57.500000   
50%                                   154.000000   
75%                                   271.500000   
max                                  1586.000000   

       Rates.Alcohol.Use Disorder Past Year.12-17  \
count                                  867.000000   
mean                                     0.040950   
std                                      0.018241   
min                                      0.012143   
25%                                      0.025503   
50%                                      0.038740   
75%                                      0.053900   
max                                      0.112131   

       Rates.Alcohol.Use Disorder Past Year.18-25  \
count                                  867.000000   
mean                                     0.151180   
std                                      0.038558   
min                                      0.071218   
25%                                      0.120067   
50%                                      0.148268   
75%                                      0.177735   
max                                      0.272941   

       Rates.Alcohol.Use Disorder Past Year.26+  ...  \
count                                867.000000  ...   
mean                                   0.059768  ...   
std                                    0.009946  ...   
min                                    0.033838  ...   
25%                                    0.053102  ...   
50%                                    0.059553  ...   
75%                                    0.065754  ...   
max                                    0.112720  ...   

       Totals.Marijuana.Used Past Year.26+  \
count                           867.000000   
mean                            348.762399   
std                             449.851031   
min                              17.000000   
25%                              91.500000   
50%                             212.000000   
75%                             439.500000   
max                            4476.000000   

       Rates.Marijuana.Used 